### PR TITLE
Ovl_Obj_Tokei_Step OK

### PR DESCRIPTION
**Pull request by hensldm**
Struct variables were determined based on experience in kz.

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h [hunk 1]
```diff
@@ -5,9 +5,20 @@
...
 
+typedef void (*ObjTokeiStepFunc)(struct ObjTokeiStep*, struct GlobalContext*);
+
+typedef struct ObjTokeiStepStep {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ f32 posChangeY;
+    /* 0x10 */ s16 startFallingTimer; // Timer for when a step starts falling after the previous step has started bouncing during opening cutscene
+    /* 0x12 */ s8 numBounces; // Counter for times the step falls below its final Y pos, and thus will bounce back up during opening cutscene
+    /* 0x13 */ s8 hasSoundPlayed;
+} ObjTokeiStepStep; // size = 0x14
```
Comment (kyleburnette) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h:16]: I can't think of a better name off the top of my head, but I think this name has the potential to be confusing

**Review by kyleburnette — CHANGES_REQUESTED** [review]
Great job on this!

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h (hunk 1 above)
Comment (hensldm) [reply src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h:16]: I debated about this but just couldn't come up with a better name. Possibly `ObjTokeiStepPanel`. Anyone else got any ideas?

**Review by AngheloAlf — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h (hunk 1 above)
Comment (Kenix3) [reply src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h:18]: `ObjTokeiStepPanel` is better than `ObjTokeiStepStep`

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h [hunk 2]
```diff
@@ -5,9 +5,22 @@
...
+                              // during opening cutscene
+    /* 0x13 */ s8 hasSoundPlayed;
+} ObjTokeiStepStep; // size = 0x14
+
 typedef struct ObjTokeiStep {
-    /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0xA8];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15c */ ObjTokeiStepFunc actionFunc;
+    /* 0x160 */ ObjTokeiStepStep steps[7];
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.h:23]: `panels` to go with the naming theme

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c [hunk 3]
```diff
@@ -21,44 +36,260 @@ const ActorInit Obj_Tokei_Step_InitVars = {
...
+        SysMatrix_MultiplyVector3fByState(&sp4c, &step->pos);
+        step->posChangeY = 0.0f;
+        step->numBounces = 0;
+    }
+}
+
+void ObjTokeiStep_InitStepsOpen(ObjTokeiStep* this) {
+    s32 i;
+    ObjTokeiStepStep* step;
+    Vec3f sp44;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c:121]: This is some kind of `panelOffset`

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c [hunk 4]
```diff
@@ -21,44 +36,260 @@ const ActorInit Obj_Tokei_Step_InitVars = {
...
+        dustSpawnPos.z += step->pos.z;
+        func_800B1210(globalCtx, &dustSpawnPos, &D_801D15B0, &D_80AD6658, (s32)((Rand_ZeroOne() * 40.0f) + 80.0f),
+                      (s32)((Rand_ZeroOne() * 20.0f) + 50.0f));
+    }
+}
+
+void ObjTokeiStep_InitSteps(ObjTokeiStep* this) {
+    s32 i;
+    ObjTokeiStepStep* step;
+    Vec3f sp4c;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c:101]: This is some kind of `panelOffset`

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c [hunk 5]
```diff
@@ -21,44 +36,260 @@ const ActorInit Obj_Tokei_Step_InitVars = {
...
+    Quake_SetSpeed(quake, 0x4E20);
+    Quake_SetQuakeValues(quake, 1, 0, 0, 0);
+    Quake_SetCountdown(quake, 7);
+    func_8013ECE0(this->dyna.actor.xyzDistToPlayerSq, 0x78, 0x14, 0xA);
+}
+
+void ObjTokeiStep_SpawnDust(ObjTokeiStep* this, ObjTokeiStepStep* step, GlobalContext* globalCtx) {
+    s32 i;
+    s32 pad;
+    Vec3f sp84;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c:80]: This is something like `dustSpawnOffset`.

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c [hunk 6]
```diff
@@ -21,44 +36,260 @@ const ActorInit Obj_Tokei_Step_InitVars = {
...
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Tokei_Step_0x80AD5BB0/func_80AD5BB0.asm")
+extern Gfx D_06000088[];
+
+extern CollisionHeader D_06000968;
+
+static f32 D_80AD6620[] = { -105.0f, -90.0f, -75.0f, -60.0f, -45.0f, -30.0f, -15.0f };
+
+static f32 D_80AD663C[] = { -60.0f, -40.0f, -20.0f, 0.0f, 20.0f, 40.0f, 60.0f };
+
+static Vec3f D_80AD6658 = { 0.0f, 0.3f, 0.0f };
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c:48]: This is `dustEffectAccel`

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c [hunk 7]
```diff
@@ -21,44 +36,260 @@ const ActorInit Obj_Tokei_Step_InitVars = {
...
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Tokei_Step_0x80AD5BB0/func_80AD5BB0.asm")
+extern Gfx D_06000088[];
+
+extern CollisionHeader D_06000968;
+
+static f32 D_80AD6620[] = { -105.0f, -90.0f, -75.0f, -60.0f, -45.0f, -30.0f, -15.0f };
+
+static f32 D_80AD663C[] = { -60.0f, -40.0f, -20.0f, 0.0f, 20.0f, 40.0f, 60.0f };
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c:46]: `dustSpawnXOffsets`

File: src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c [hunk 8]
```diff
@@ -21,44 +36,260 @@ const ActorInit Obj_Tokei_Step_InitVars = {
...
     (ActorFunc)ObjTokeiStep_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Tokei_Step_0x80AD5BB0/func_80AD5BB0.asm")
+extern Gfx D_06000088[];
+
+extern CollisionHeader D_06000968;
+
+static f32 D_80AD6620[] = { -105.0f, -90.0f, -75.0f, -60.0f, -45.0f, -30.0f, -15.0f };
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Obj_Tokei_Step/z_obj_tokei_step.c:44]: `panelXOffsets`

**Review by Kenix3 — CHANGES_REQUESTED** [review]
Just a few documentation related things to clean up. Good work!

**Pull request merged**